torch/_inductor/select_algorithm.py

```diff
@@ -1201,19 +1389,51 @@ def make_extra() -> str:
             V.graph.set_current_device(layout.device),
             make_kernel() as kernel,
         ):
-            result = generate_code(kernel)
-            if not result:  # happens at ZeroDivisionError:
-                return None
-            code, extra = result
+            cache_entry = self._generated_code_cache.get_entry(cache_key)
+            cache_hit = False
+
+            if cache_entry is not None:
+                code, extra, events = cache_entry
+                kernel.replay_cached_events(events)
+                cache_hit = True
+
+            else:
+                result = generate_code(kernel)
+                if result is None:  # happens at ZeroDivisionError:
+                    return None
+                code, extra = result
+                self._generated_code_cache.put_entry(
+                    cache_key, code, extra, kernel.cached_replay_events
+                )
 
         assert code is not None and extra is not None
 
+        if cache_hit and self.test_cache:
+            with (
```

Comment:
nit: factor this out to helper function, or as optoinal hook to avoid putting the test logic inline in main logic

Reply:
done